runtime/tuple.go

```diff
@@ -75,6 +75,14 @@ func tupleContains(f *Frame, t, v *Object) (*Object, *BaseException) {
 	return seqContains(f, t, v)
 }
 
+func tupleCount(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	argc := len(args)
+	if argc != 2 {
+		return nil, f.RaiseType(TypeErrorType, fmt.Sprintf("count() takes exactly one argument (%d given)", argc))
```

Comment:
Agree about doing the same everywhere. And you're right that the common case will be caught by the method type check you mention. The trickier case is if somebody gets the underlying function via instancemethod's im_func attribute:

```
>>> class A(object):
...  def foo(self):
...   pass
...
>>> A.foo
<unbound method A.foo>
>>> A.foo.im_func
<function foo at 0x10407f140>
```

Note that its not possible to get the underlying function for tuple.count in CPython because builtin methods are not instancemethod objects there:

```
>>> tuple.count
<method 'count' of 'tuple' objects>
>>> tuple.count.im_func
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'method_descriptor' object has no attribute 'im_func'
```

However, in Grumpy (at least for the time being), builtin methods *are* instancemethods. So it's possible for somebody to get the underlying function for tuple.count and call that directly. So the redundant check is actually important here.